tests/framework/extensions/sshkeys/downloadsshkeys.go

```diff
@@ -10,41 +10,41 @@ import (
 )
 
 const (
-	privateKeySSHKeyRegExPattern              = `----BEGIN RSA PRIVATE KEY-{3,}\n([\s\S]*?)\n-{3,}END RSA PRIVATE KEY-----`
+	privateKeySSHKeyRegExPattern              = `-----BEGIN RSA PRIVATE KEY-{3,}\n([\s\S]*?)\n-{3,}END RSA PRIVATE KEY-----`
 	ClusterMachineConstraintResourceSteveType = "cluster.x-k8s.io.machine"
 )
 
 // DownloadSSHKeys is a helper function that takes a client, the machinePoolNodeName to download
 // the ssh key for a particular node.
-func DownloadSSHKeys(client *rancher.Client, machinePoolNodeName string) (string, error) {
+func DownloadSSHKeys(client *rancher.Client, machinePoolNodeName string) ([]byte, error) {
```

Comment:
@markusewalker I kind of remember not wanting to do this. Will this be ok?

Reply:
If I remember correctly, I think that tied into Azure issues though, right? I think this should be okay, at least there were no apparent failures in the freeform jobs indicating otherwise